### ISSUE TITLE
no-unused-prop-types does not process intersection flow types

### ISSUE THREAD

**Issue by mruzekw** [yannickcr/eslint-plugin-react#1323]
`no-unused-prop-types` does not currently flag unused PropTypes in flow intersection types.

In this case, it should be flagging `age`.

```
type AProps = {
  name: string;
  age: number;
};

type BProps = {
  greeting: string;
};

type Props = AProps & BProps;

class Hello extends React.Component {
  props: Props;

  render () {
    return <div>{this.props.greeting}, {this.props.name}</div>;
  }
}
```

**Comment by mruzekw** [comment]
Looks like this is being addressed in https://github.com/yannickcr/eslint-plugin-react/pull/1404
